src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -231,10 +232,22 @@ private boolean areTotalsEqual(InvoiceLine invoiceLine1, InvoiceLine invoiceLine
       }));
   }
 
+  /**
+   * Deletes Invoice Line and update Invoice if deletion is allowed
+   * 1. Get invoiceLine by id
+   * 2. Get invoice by id
+   * 3. Verify if user has permission to delete invoiceLine based on acquisitions units, if not then return
+   * 4. If user has permission to delete then delete invoiceLine
+   * 5. Update corresponding Invoice
+   * @param id invoiceLine id to be deleted
+   */
   public CompletableFuture<Void> deleteInvoiceLine(String id) {
     return getInvoiceLine(id)
-      .thenCompose(line -> handleDeleteRequest(resourceByIdPath(INVOICE_LINES, id, lang), httpClient, ctx, okapiHeaders, logger)
-        .thenRun(() -> updateInvoiceAndLinesAsync(line.getInvoiceId())));
+      .thenCompose(invoiceLine -> getInvoiceById(invoiceLine.getInvoiceId(), lang, httpClient, ctx, okapiHeaders, logger))
```

Comment:
Eventually I think we're going to have the ability to search of invoices by invoiceLine fields, I'm not sure that's in place yet.  If not we do need to get the invoiceLine first, but this step can be optimized away once that feature is implemented

Reply:
I believe that `GET /invoice/invoices?query=invoice_lines.id==<id>` can be use here e.g. `GET /invoice/invoices?query=invoice_lines.id==0155fee2-9b2d-4020-9b91-060d0d102209`.

Reply:
Do we have that hooked up already? I wasn't sure

Reply:
Ah, is this it here?
  https://github.com/folio-org/mod-invoice-storage/blob/8abb4c98f5a99f0758966bdfd0c7da798d7b4fd5/src/main/resources/templates/db_scripts/schema.json#L13

Reply:
yes, this is [foreign key cross table index queries](https://github.com/folio-org/raml-module-builder/tree/master#cql2pgjson-foreign-key-cross-table-index-queries) feature introduced in RMB recently